packages/app-store/office365calendar/lib/CalendarService.ts

```diff
@@ -260,8 +260,11 @@ export default class Office365CalendarService implements Calendar {
         credential.userId
       );
       const responseJson = await handleErrorsJson(response);
-      const tokenResponse = parseRefreshTokenResponse(responseJson, refreshTokenResponseSchema);
-      o365AuthCredentials = { ...o365AuthCredentials, ...(tokenResponse.success && tokenResponse.data) };
```

Comment:
Notice how other apps handle the safe parse return in the app package

Reply:
parseRefreshTokenResponse has nothing to do with the type; auto-infer does that; so for o365calendar you can do the following to make it work similarly.

```typescript
const tokenResponse: z.infer<typeof refreshTokenResponseSchema> = refreshTokenResponseSchema(responseJson, refreshTokenResponseSchema);
```